systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java

```diff
@@ -20,7 +20,7 @@ public String getKind() {
     }
     @Override
     public Deployment get(String namespace, String name) {
-        return ResourceManager.kubeClient().namespace(namespace).getDeployment(name);
+        return ResourceManager.kubeClient().namespace(namespace).getDeployment(namespace, name);
```

Comment:
Does it point to some problem in fabric8? Or why do we need to specify namespace twice?

Reply:
No, but in the case of parallel execution, there is a potential error to have a race condition. Because this piece of code is not synchronized, thus multiple threads could access it (with different namespaces) and it can change the `shared` state of the Kube client. Thus we need it explicitly specify `namespace` inside `getDeployment(...)` method.

Reply:
What I can do about it is remove `unnecessary` switching between namespaces from:
```java
return ResourceManager.kubeClient().namespace(namespace).getDeployment(namespace, name);
```
to
```java
return ResourceManager.kubeClient().getDeployment(namespace, name);
```

Reply:
Ye, that change would be nice.

Reply:
Changed )